GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -578,6 +581,54 @@ protected override bool ProcessCmdKey(ref Message msg, Keys keyData)
             return base.ProcessCmdKey(ref msg, keyData);
         }
 
+        public void LoadCustomDifftools()
+        {
+            openWithDifftoolToolStripMenuItem.DropDown = null;
+            stagedOpenDifftoolToolStripMenuItem9.DropDown = null;
+            if (!AppSettings.ShowAllCustomDiffTools)
+            {
+                return;
+            }
+
+            ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                var difftool = Module.GetEffectiveSetting(SettingKeyString.DiffToolKey);
+                var tools = await Module.GetCustomDiffMergeTools(isDiff: true);
+                if (tools.Count<string>() <= 1)
+                {
+                    // No need to show the menu
+                    return;
+                }
+
+                openWithDifftoolToolStripMenuItem.DropDown = new ContextMenuStrip(components);
+                stagedOpenDifftoolToolStripMenuItem9.DropDown = new ContextMenuStrip(components);
+                foreach (var tool in tools)
+                {
+                    var unstagedItem = new ToolStripMenuItem(tool) { Tag = tool };
+                    var stagedItem = new ToolStripMenuItem(tool) { Tag = tool };
+
+                    unstagedItem.Click += UnstagedOpenWithDifftoolToolStripMenuItem_Click;
+                    stagedItem.Click += StagedOpenDifftoolToolStripMenuItem_Click;
+                    if (tool == difftool)
+                    {
+                        unstagedItem.Font = new Font(unstagedItem.Font, FontStyle.Bold);
+                        stagedItem.Font = new Font(stagedItem.Font, FontStyle.Bold);
+
+                        unstagedItem.ShortcutKeyDisplayString = openWithDifftoolToolStripMenuItem.ShortcutKeyDisplayString;
+                        stagedItem.ShortcutKeyDisplayString = stagedOpenDifftoolToolStripMenuItem9.ShortcutKeyDisplayString;
+
+                        openWithDifftoolToolStripMenuItem.DropDown.Items.Insert(0, unstagedItem);
+                        stagedOpenDifftoolToolStripMenuItem9.DropDown.Items.Insert(0, stagedItem);
+                    }
+                    else
+                    {
+                        openWithDifftoolToolStripMenuItem.DropDown.Items.Add(unstagedItem);
+                        stagedOpenDifftoolToolStripMenuItem9.DropDown.Items.Add(stagedItem);
+                    }
+                }
+            }).FileAndForget();
```

Comment:
We have a somewhat similar code repeated in several places... Can we generalise it somehow?